rules/import_middleman.bzl

```diff
@@ -260,23 +264,61 @@ def _file_collector_rule_impl(ctx):
         **objc_provider_fields
     )
 
-    additional_providers = []
+    # Create the deduplicated CcInfo provider information
+    static_libraries = objc_provider_fields.get("library", depset([])).to_list() + \
+                       objc_provider_fields.get("static_framework_file", depset([])).to_list() + \
+                       objc_provider_fields.get("imported_library", depset([])).to_list()
+    force_load_static_libraries = objc_provider_fields.get("force_load_library", depset([])).to_list()
+    dynamic_libraries = objc_provider_fields.get("dynamic_framework_file", depset([])).to_list()
```

Comment:
Could we merge these as `depset`s instead of dropping down to list?  This could change how duplicates are handled though

Reply:
We have to iterate on them below to create the `LibraryToLink` so don't think we can. I tried to make the bare minimum changes to this rule to get CI passing. 

I imagine were still dropping linking information / not deduping the inputs in Bazel 7+ because we the list is using information from `ObjcProvider`. But this rule's code is extremely confusing to me, so tried to keep it minimal. I'm hoping one of the Square folks can revisit this if it's required for other Bazel 7+ changes.

Reply:
I ended up reverting most of the changes here, it was regressing the build in Bazel 6 for Square folks. These current changes are enough to pass CI here in both 6 and 7 and fix that regression. I imagine another pass will be needed in this rule for Bazel 7 in other use cases though